### DISCUSSION TITLE
support  set routes config match cookie value ?

### DISCUSSION THREAD

**Discussion by iangithub** [dotnet/yarp#1582] [Q&A] [answered]
Can i set routes config match cookie value , like traefik HeadersRegexp("Cookie",'ap-ver=1.0') ?

eg : cookie ap-ver=1.0 go to 192.168.1.1 , and cookie ap-ver=2.0 go to 192.168.1.2

**Comment by greenEkatherine** [comment, marked as answer]
Yes, what you need is header routing based on `HeaderPrefix` or `Contains` modes https://microsoft.github.io/reverse-proxy/articles/header-routing.html#configuration:
```json  "route2" : {
    "ClusterId": "cluster1",
    "Match": {
      "Path": "{**catch-all}",
      "Headers": [
        {
          "Name": "Cookie",
          "Values": [ "ap-ver=1.0" ],
          "Mode": "HeaderPrefix"
        }
      ]
    }
  }
  ```

**Reply by iangithub** [reply]
Thank you 👍

**Reply by Tratcher** [reply]
Also read through https://github.com/microsoft/reverse-proxy/issues/1494
